membership_variable_period/tests/test_membership_variable_period.py

```diff
@@ -112,33 +123,34 @@ def test_create_and_modify_invoice_line_membership_product(self):
         self.product.membership_interval_unit = 'days'
         invoice_line = self.env['account.invoice.line'].create(
             {'product_id': self.product.id,
+             'price_unit': 100,
+             'account_id': self.partner_account,
              'name': 'Membership w/o prorrate'})
         invoice = self.env['account.invoice'].create(
             {'partner_id': self.partner.id,
              'date_invoice': '2015-07-01',
-             'account_id': self.partner.property_account_receivable.id})
+             'account_id': self.partner_account})
         invoice_line.invoice_id = invoice.id
-        membership_line = invoice.invoice_line[0].membership_lines[0]
+        membership_line = invoice_line[0].membership_lines[0]
         self.assertEqual(membership_line.date_from, '2015-07-01')
         self.assertEqual(membership_line.date_to, '2015-07-20')
         self.assertEqual(self.partner.membership_start, '2015-07-01')
         self.assertEqual(self.partner.membership_stop, '2015-07-20')
 
     def test_check_membership_expiry(self):
-        self.env['membership.membership_line'].create(
-            {'partner': self.partner.id,
-             'membership_id': self.product.id,
-             'member_price': 1.0,
-             'date': '2014-01-01',
-             'date_from': '2014-01-01',
-             'date_to': '2014-12-31',
-             'state': 'paid',
-             'account_invoice_line': self.env['account.invoice.line'].search(
-                 [], limit=1).id})
-        # Force state to let the calculation return to the computed one
+        self.product.membership_interval_unit = 'years'
+        self.env['account.invoice'].create(
+            {'partner_id': self.partner.id,
+             'date_invoice': '2014-07-01',  # It's leap year
+             'account_id': self.partner_account,
+             'invoice_line_ids': [(0, 0, {'product_id': self.product.id,
+                                          'price_unit': 100,
+                                          'account_id': self.partner_account,
+                                          'name': 'Membership wth prorrate'})]}
+        )
         self.partner.membership_stop = '2014-12-31'
         self.env['res.partner'].check_membership_expiry()
-        self.assertEqual(self.partner.membership_state, 'none')
+        self.assertEqual(self.partner.membership_state, 'old')
```

Comment:
@pedrobaeza Hum, i have to check https://github.com/OCA/vertical-association/pull/21